website/data/access-tokens/create-access-token.ts

```diff
@@ -0,0 +1,53 @@
+import { PostgrestError } from "@supabase/supabase-js"
+import { useMutation, UseMutationOptions, useQueryClient } from "@tanstack/react-query"
+import supabase from "~/lib/supabase"
+
+type NewAccessTokenVariables = {
+    tokenName: string
+}
+
+export async function newAccessToken({
+  tokenName,
+}: NewAccessTokenVariables) {
+  const { data, error } = await supabase.rpc('new_access_token', {
+    token_name: tokenName,
+  })
+
+  if (error) throw error
+  return data
+}
+
+type NewAccessTokenData = Awaited<ReturnType<typeof newAccessToken>>
+type NewAccessTokenError = PostgrestError
+
+export const useNewAccessTokenMutation = ({
+  onSuccess,
+  ...options
+}: Omit<
+  UseMutationOptions<
+    NewAccessTokenData,
+    NewAccessTokenError,
+    NewAccessTokenVariables
+  >,
+  'mutationFn'
+> = {}) => {
+  const queryClient = useQueryClient()
+
+  return useMutation<
+    NewAccessTokenData,
+    NewAccessTokenError,
+    NewAccessTokenVariables
+  >(
+    ({ tokenName }) =>
+      newAccessToken({ tokenName }),
+    {
+      async onSuccess(data, variables, context) {
+        await Promise.all([
+          queryClient.resetQueries(),
```

Comment:
Is there any reason you're using `resetQueries` here instead of just invalidating the `['access-tokens']` query?
For example:
```typescript
queryClient.invalidateQueries(['access-tokens'])
```

Reply:
It was a copy paste mistake. I've changed it to invalidate only the 'access-tokens' query.